client/client-http/src/main/java/software/amazon/smithy/java/client/http/compression/GzipCompressingInputStream.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package software.amazon.smithy.java.client.http.compression;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.zip.GZIPOutputStream;
+
+final class GzipCompressingInputStream extends InputStream {
```

Comment:
Add javadoc.

Reply:
This implementation underneath still reads the whole input stream into memory to do the compression.